src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -449,6 +449,12 @@ class SchedulerActions(
         val knownTaskStatuses = appIds.flatMap { appId =>
           taskTracker.get(appId).collect {
             case task: Protos.MarathonTask if task.hasStatus => task.getStatus
+            case task: Protos.MarathonTask if task.hasSlaveId => // staged tasks without status but with slaveId
```

Comment:
Great patch, @janisz!

I would prefer if stored the builder in a var:

`val builder = TaskStatus.newBuilder` ...
and had an `if (task.hasSlaveId)` that adds the slaveId. That makes it more obvious that this case only differs in the slaveId. Now we have two nearly identical code blocks...